### PR TITLE
Pass imagePullSecrets as operator flag

### PR DESCRIPTION
**Pull request by hossainemruz**
Signed-off-by: hossainemruz <emruz@appscode.com>

### CODE REVIEW

File: charts/stash-enterprise/templates/deployment.yaml [hunk 1]
```diff
@@ -46,6 +46,13 @@ spec:
         - --v={{ .Values.logLevel }}
         - --docker-registry={{ .Values.operator.registry }}
         - --image-tag={{ .Values.operator.tag }}
+        {{ if .Values.imagePullSecrets }}
```
Comment (tamalsaha) [inline charts/stash-enterprise/templates/deployment.yaml:49]: Just use https://github.com/Masterminds/sprig/blob/master/docs/string_slice.md#join

{{ with .Values.imagePullSecrets }}
- --image-pull-secrets={{ join "," . }}
{{ end }}

Comment (hossainemruz) [reply charts/stash-enterprise/templates/deployment.yaml:49]: It will not work. `.Values.imagePullSecrets` is a map. Not slice.

Comment (tamalsaha) [reply charts/stash-enterprise/templates/deployment.yaml:49]: Right! We can get the values in a list using `values` https://github.com/Masterminds/sprig/blob/master/docs/dicts.md#values

`{{ values . | join "," }}`

File: charts/stash-enterprise/templates/cluster-role.yaml [hunk 2]
```diff
@@ -67,7 +67,7 @@ rules:
   - secrets
   - services
   - endpoints
-  verbs: ["get"]
+  verbs: ["get","list","create","patch"]
```
Comment (tamalsaha) [inline charts/stash-enterprise/templates/cluster-role.yaml:70]: Why do we need this?

Comment (hossainemruz) [reply charts/stash-enterprise/templates/cluster-role.yaml:70]: We create the docker secret from the operator in the workload namespace.

Comment (tamalsaha) [reply charts/stash-enterprise/templates/cluster-role.yaml:70]: Let's separate our the secrets part and just give the extra permission for secrets, not for services and endpoints.

Comment (hossainemruz) [reply charts/stash-enterprise/templates/cluster-role.yaml:74]: done

File: charts/stash-enterprise/templates/mutating-webhook.yaml [hunk 3]
```diff
@@ -140,4 +140,25 @@ webhooks:
 {{- if and (ge $major 1) (ge $minor 12) }}
   sideEffects: None
 {{- end }}
+- name: restoresession.admission.stash.appscode.com
```
Comment (tamalsaha) [inline charts/stash-enterprise/templates/mutating-webhook.yaml:143]: What are we mutating?

Comment (hossainemruz) [reply charts/stash-enterprise/templates/mutating-webhook.yaml:143]: We are migrating `RestoreSession`.

https://github.com/stashed/apimachinery/blob/b8c42c18d61fed4cb4ed7dde6b9ba496066636b2/apis/stash/v1beta1/restore_session_helpers.go#L51

**Pull request merged**